diff --git a/src/include/86box/machine.h b/src/include/86box/machine.h
index e8b29f800..4059ff87c 100644
--- a/src/include/86box/machine.h
+++ b/src/include/86box/machine.h
@@ -503,6 +503,10 @@ extern const device_t  ibmat_device;
 #endif
 extern int             machine_at_ibmat_init(const machine_t *);
 #ifdef EMU_DEVICE_H
+extern const device_t  olivetti_pcs286s_device;
+#endif
+extern int             machine_at_olivetti_pcs286s_init(const machine_t *);
+#ifdef EMU_DEVICE_H
 extern const device_t  ibmxt286_device;
 #endif
 extern int             machine_at_ibmxt286_init(const machine_t *);
diff --git a/src/machine/m_at_286.c b/src/machine/m_at_286.c
index 291ff8e0c..a6a45e830 100644
--- a/src/machine/m_at_286.c
+++ b/src/machine/m_at_286.c
@@ -137,6 +137,53 @@ const device_t ibmat_device = {
     .config        = ibmat_config
 };
 
+static const device_config_t olivetti_pcs286s_config[] = {
+    // clang-format off
+    {
+        .name           = "bios",
+        .description    = "BIOS Version",
+        .type           = CONFIG_BIOS,
+        .default_string = "olivetti_pcs286s_106",
+        .default_int    = 0,
+        .file_filter    = NULL,
+        .spinner        = { 0 },
+        .selection      = { { 0 } },
+        .bios           = {
+            {
+                .name          = "Rel. 1.06",
+                .internal_name = "olivetti_pcs286s_106",
+                .bios_type     = BIOS_NORMAL,
+                .files_no      = 2,
+                .local         = 0,
+                .size          = 65536,
+                .files         = {
+                    "roms/machines/olivetti_pcs286s/PCS286S_REL.1.06_LOW.BIN",
+                    "roms/machines/olivetti_pcs286s/PCS286S_REL.1.06_HIGH.BIN",
+                    ""
+                }
+            },
+
+            { .files_no = 0 }
+        },
+    },
+    { .name = "", .description = "", .type = CONFIG_END }
+    // clang-format on
+};
+
+const device_t olivetti_pcs286s_device = {
+    .name          = "Olivetti PCS 286S",
+    .internal_name = "olivetti_pcs286s",
+    .flags         = 0,
+    .local         = 0,
+    .init          = NULL,
+    .close         = NULL,
+    .reset         = NULL,
+    .available     = NULL,
+    .speed_changed = NULL,
+    .force_redraw  = NULL,
+    .config        = olivetti_pcs286s_config
+};
+
 static void
 machine_at_ibm_common_init(const machine_t *model)
 {
@@ -984,6 +1031,35 @@ machine_at_scat_init(const machine_t *model, int is_v4, int is_ami)
         device_add(&scat_device);
 }
 
+int
+machine_at_olivetti_pcs286s_init(const machine_t *model)
+{
+    int         ret;
+    const char *fn[2];
+
+    if (!device_available(model->device))
+        return 0;
+
+    device_context(model->device);
+    fn[0] = device_get_bios_file(model->device, device_get_config_bios("bios"), 0);
+    fn[1] = device_get_bios_file(model->device, device_get_config_bios("bios"), 1);
+    ret   = bios_load_interleaved(fn[0], fn[1], 0x000e0000, 131072, 0);
+    device_context_restore();
+
+    if (bios_only || !ret)
+        return ret;
+
+    machine_at_scat_init(model, 0, 1);
+
+    if (fdc_current[0] == FDC_INTERNAL)
+        device_add(&fdc_at_device);
+
+    device_add(&f82c710_device);
+    device_add(&ide_isa_device);
+
+    return ret;
+}
+
 int
 machine_at_pc5286_init(const machine_t *model)
 {
diff --git a/src/machine/machine_table.c b/src/machine/machine_table.c
index 08f386e19..88872a554 100644
--- a/src/machine/machine_table.c
+++ b/src/machine/machine_table.c
@@ -4937,6 +4937,54 @@ const machine_t machines[] = {
         .net_device               = NULL,
         .aliases                  = { "Amstrad VSC286E", "" }
     },
+    /* Uses Olivetti AT KBC firmware. */
+    {
+        .name              = "[SCAT] Olivetti PCS 286S",
+        .internal_name     = "olivetti_pcs286s",
+        .type              = MACHINE_TYPE_286,
+        .chipset           = MACHINE_CHIPSET_SCAT,
+        .init              = machine_at_olivetti_pcs286s_init,
+        .p1_handler        = machine_generic_p1_handler,
+        .gpio_handler      = NULL,
+        .available_flag    = MACHINE_AVAILABLE,
+        .gpio_acpi_handler = NULL,
+        .cpu               = {
+            .package     = CPU_PKG_286,
+            .block       = CPU_BLOCK_NONE,
+            .min_bus     = 8000000,
+            .max_bus     = 16000000,
+            .min_voltage = 0,
+            .max_voltage = 0,
+            .min_multi   = 0,
+            .max_multi   = 0
+        },
+        .bus_flags = MACHINE_AT,
+        .flags     = MACHINE_IDE,
+        .ram       = {
+            .min  = 512,
+            .max  = 4096,
+            .step = 512
+        },
+        .nvrmask                  = 127,
+        .jumpered_ecp_dma         = 0,
+        .default_jumpered_ecp_dma = -1,
+        .kbc_device               = &kbc_at_device,
+        .kbc_params               = KBC_VEN_OLIVETTI,
+        .nvr_device               = &nvr_at_device,
+        .nvr_params               = NVR_AT,
+        .sio_device               = NULL,
+        .sio_params               = 0x00000000,
+        .kbc_p1                   = 0x000004f0,
+        .gpio                     = 0xffffffff,
+        .gpio_acpi                = 0xffffffff,
+        .device                   = &olivetti_pcs286s_device,
+        .kbd_device               = NULL,
+        .fdc_device               = NULL,
+        .vid_device               = NULL,
+        .snd_device               = NULL,
+        .net_device               = NULL,
+        .aliases                  = { "" }
+    },
     /* Has unknown KBC firmware. */
     {
         .name              = "[SCAT] Amstrad PC5286",
